kratos/tests/test_linear_constraints.py

```diff
@@ -19,98 +21,90 @@ def setUp(self):
-        KratosMultiphysics.VariableUtils().SetVariable(
-            KratosMultiphysics.DISPLACEMENT_X, 0.0, bcs.Nodes)
-        KratosMultiphysics.VariableUtils().SetVariable(
-            KratosMultiphysics.DISPLACEMENT_Y, 0.0, bcs.Nodes)
+        KM.VariableUtils().SetVariable(KM.DISPLACEMENT_X, 0.0, bcs.Nodes)
+        KM.VariableUtils().SetVariable(KM.DISPLACEMENT_Y, 0.0, bcs.Nodes)
 
-        KratosMultiphysics.VariableUtils().ApplyFixity(
-            KratosMultiphysics.DISPLACEMENT_X, True, bcs.Nodes)
-        KratosMultiphysics.VariableUtils().ApplyFixity(
-            KratosMultiphysics.DISPLACEMENT_Y, True, bcs.Nodes)
+        KM.VariableUtils().ApplyFixity(KM.DISPLACEMENT_X, True, bcs.Nodes)
+        KM.VariableUtils().ApplyFixity(KM.DISPLACEMENT_Y, True, bcs.Nodes)
 
         bcmn = self.mp.GetSubModelPart("MovingNodes")
-        KratosMultiphysics.VariableUtils().SetVariable(
-            KratosMultiphysics.DISPLACEMENT_X, 0.01, bcmn.Nodes)
-        KratosMultiphysics.VariableUtils().SetVariable(
-            KratosMultiphysics.DISPLACEMENT_Y, 0.0, bcmn.Nodes)
-        KratosMultiphysics.VariableUtils().ApplyFixity(
-            KratosMultiphysics.DISPLACEMENT_X, True, bcmn.Nodes)
-        KratosMultiphysics.VariableUtils().ApplyFixity(
-            KratosMultiphysics.DISPLACEMENT_Y, True, bcmn.Nodes)
+        KM.VariableUtils().SetVariable(KM.DISPLACEMENT_X, 0.01, bcmn.Nodes)
+        KM.VariableUtils().SetVariable(KM.DISPLACEMENT_Y, 0.0, bcmn.Nodes)
+        KM.VariableUtils().ApplyFixity(KM.DISPLACEMENT_X, True, bcmn.Nodes)
+        KM.VariableUtils().ApplyFixity(KM.DISPLACEMENT_Y, True, bcmn.Nodes)
 
-    def _setup_solver(self):
+    def _setup_solver(self, solving_with = "Block", linear_solver = "AMGCL"):
 
         #define a minimal newton raphson solver
-        self.linear_solver = KratosMultiphysics.SkylineLUFactorizationSolver()
-        self.builder_and_solver = KratosMultiphysics.ResidualBasedBlockBuilderAndSolver(
-            self.linear_solver)
-        self.scheme = KratosMultiphysics.ResidualBasedBossakDisplacementScheme(
-            -0.01)
-        self.convergence_criterion = KratosMultiphysics.ResidualCriteria(
-            1e-10, 1e-12)
+        if linear_solver == "AMGCL":
+            self.linear_solver = KM.AMGCLSolver()
+        else:
+            self.linear_solver = KM.SkylineLUFactorizationSolver()
+        if solving_with == "Block":
+            self.builder_and_solver = KM.ResidualBasedBlockBuilderAndSolver(self.linear_solver)
+        elif solving_with == "LM":
+            params = KM.Parameters("""{
+                "diagonal_values_for_dirichlet_dofs"                 : "use_max_diagonal",
+                "silent_warnings"                                    : false,
+                "consider_lagrange_multiplier_constraint_resolution" : "Single"
+            }""")
+            self.builder_and_solver = KM.ResidualBasedBlockBuilderAndSolverWithLagrangeMultiplier(self.linear_solver, params)
```

Comment:
i would put this in a different test. It is better to have many small test files than few large ones

Reply:
This is the best way to avoid code duplication, it only changes the B&S

Reply:
please do split it. My point is that the current version of  it is much easier to understand than after your modification. In short i definitely would prefer two tests i understand that one that i don't...

Reply:
In https://github.com/KratosMultiphysics/Kratos/pull/6789 I separate the modification of the test. The quantity of code added to the test **is only 4 lines after that is merged**. Reducing code duplication is one of the best way to avoid errors, we see that constantly each time we do an important change in the API and we refactor code and we see like the same error was done several times